xbitinfo/xbitinfo.py

```diff
@@ -35,6 +35,59 @@ def get_user_input():
     return args
 
 
+def get_bit_coords(dtype_size):
+    """Get coordinates for bits assuming float dtypes."""
+    if dtype_size == 16:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 6)]
+            + [f"m{int(i-5)}" for i in range(6, 16)]
+        )
+    elif dtype_size == 32:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 9)]
+            + [f"m{int(i-8)}" for i in range(9, 32)]
+        )
+    elif dtype_size == 64:
+        coords = (
+            ["±"]
+            + [f"e{int(i)}" for i in range(1, 12)]
+            + [f"m{int(i-11)}" for i in range(12, 64)]
+        )
+    else:
+        raise ValueError(f"dtype of size {dtype_size} neither known nor implemented.")
+    return coords
+
+
+def dict_to_dataset(info_per_bit):
+    """Convert keepbits dictionary to dataset."""
+    dsb = xr.Dataset()
+    for v in info_per_bit.keys():
+        dtype_size = len(info_per_bit[v])
+        dim_name = f"bit{dtype_size}"
+        dsb[v] = xr.DataArray(
+            info_per_bit[v],
+            dims=[dim_name],
+            coords={dim_name: get_bit_coords(dtype_size)},
+            name=v,
+            attrs={"long_name": f"{v} bitwise information"},
+        ).astype("float64")
+    # add metadata
+    dsb.attrs = {
+        "xbitinfo_description": "bitinformation calculated by xbitinfo.get_bitinformation wrapping bitinformation.jl",
+        "python_repository": "https://github.com/observingClouds/xbitinfo",
+        "julia_repository": "https://github.com/milankl/BitInformation.jl",
+        "reference_paper": "http://www.nature.com/articles/s43588-021-00156-2",
+    }
+    for c in dsb.coords:
+        if "bit" in c:
+            dsb.coords[c].attrs = {
+                "description": "name of the bits: '±' refers to the sign bit, 'e' to the exponents bits and 'm' to the mantissa bits."
+            }
```

Comment:
Should we add ds attrs here?

Reply:
@aaronspring, do you want to add the global attributes from the original dataset here?

Reply:
Yeah. That's the question. Or do you think this would make info_per_bit too busy? I am unsure.

Reply:
for air_temperature in the doctests it is more

Reply:
But let's assume metadata is correct usually